dev_guide/builds.adoc

```diff
@@ -912,3 +912,131 @@ case `*scmsecret*`:
 <1> The URL of private repository is usually in the form
 `git@example.com:<username>/<repository>`.
 ====
+
+
+== Limiting the type of build strategies for builds
+
+
+The build strategies that users are allowed to use when creating `*Builds*` or `*BuildConfigs*`
+can be controlled via
+link:../architecture/additional_concepts/authorization.html[authorization policy].
+Each build strategy is controlled via access to a build subresource:
+
+.Build strategy subresources
+[cols="1,1",options="header"]
+|===
+
+|Strategy |Resource
+
+|Docker
+|builds/docker
+
+|Source-To-Image
+|builds/source
+
+|Custom
+|builds/custom
+
+|===
+
+By default, project administrators (`admin` role) and project editors (`edit` role) are granted
+permission to use all strategies (Docker, Source-to-Image, and Custom).
+
+
+=== Disabling a build strategy globally
+To prevent access to a particular build strategy globally, login as cluster administrator and edit
+each of the default roles:
+
+----
+$ oadm edit clusterrole admin
+$ oadm edit clusterrole edit
+----
+
+For each role, remove the line that corresponds to the resource of the strategy to disable:
+
+.Disable the Docker build strategy
+=====
+
+----
+kind: ClusterRole
+metadata:
+  name: admin
+...
+
+rules:
+- attributeRestrictions: null
+  resources:
+  - builds/custom
+  - builds/docker <1>
+  - builds/source
+  - pods/exec
+  - pods/portforward
+
+...
+
+----
+<1> Delete this line to disable Docker builds globally
+
+=====
+
+=== Allowing only certain users to create builds with a particular strategy
+
+To allow only a set of specific users to create builds with a paraticular strategy:
+
+1. link:#disabling-a-build-strategy-globally[Remove the build strategy resource from the default admin and edit roles].
+1. Create a separate role for that build strategy
++
+[#create-separate-role]
+.Create a separate role for the docker build strategy
+====
+
+Create a file that defines the new cluster role (dockerstrategy.yaml):
+
+----
+kind: ClusterRole
+apiVersion: v1
+metadata:
+  name: dockerbuilder
+rules:
+- resources:
+  - builds/docker
+  verbs:
+  - create
+----
+
+
+As cluster administrator, create the cluster role:
+
+----
+$ oc create -f  dockerstrategy.yaml
+----
+
+====
+
+1. Assign the cluster role to an individual user
++
+.Assign role dockerbuilder to devuser
+====
+
+----
+$ oadm policy add-cluster-role-to-user dockerbuilder devuser
```

Comment:
This will grant `devuser` access to `builds/docker` in all projects, even those he can't view `builds` in.  You probably want `oadm policy add-role-to-user dockerbuilder devuser -n <project name>`